src/main/webapp/trips/tripScript.js

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Triggered upon DOM load.
+$(document).ready(() => {
+  // Redirect to homepage if user is not signed in.
+  isSignedIn().then((signInStatus) => {
+    if (!signInStatus) {
+      window.location.replace('/');
+    } else {
+      setContentWidth('800px');
+      displayHeader();
+    }
+  });
+});
+
+// Returns Promise with the sign in status in a boolean.
+function isSignedIn() {
+  return new Promise((resolve, reject) => {
+    fetch('/auth').then(response => response.json()).then((signInStatus) => {
+      resolve(signInStatus);
+    }).catch((error) => {
+      reject(error);
+    });
+  });
+}
+
+function displayHeader() {
+  // Display header for site.
+  const header = document.getElementById('header');
+  header.style.display = 'block';
+}
+
+// Set the width of the content container.
+function setContentWidth(width) {
```

Comment:
There's some repeated code here with the functions in script.js (displayHeader() and setContentWidth(width)), so we should consider restructuring the code in a way that we don't have to define these same functions in every script file. But I think this is fine for the prototype!

Reply:
Absolutely agree, I'm looking to clean up some of the repeated JS code across files next week when building the MVP. Thanks for the note, this sounds good!